lib/modules/datasource/packagist/index.ts

```diff
@@ -232,29 +191,22 @@ export class PackagistDatasource extends Datasource {
         return null;
       }
       const {
-        packages,
         providersUrl,
         providersLazyUrl,
         providerPackages,
         includesPackages,
       } = allPackages;
-      if (packages?.[packageName]) {
-        const dep = PackagistDatasource.extractDepReleases(
-          packages[packageName]
-        );
-        return dep;
-      }
```

Comment:
From what I've seen, `packages` field is always an array:
- [`packagist`](https://repo.packagist.org/packages.json)
- [`wpackagist`](https://wpackagist.org/packages.json)
- [`satis.spatie.be`](https://satis.spatie.be/packages.json)

It means, we never get to this branch as far as `packageName` is string.

Reply:
maybe some custom old private registries?

Reply:
I don't know, packagist protocol is poorly documented. Do you think we need to keep this implementation?

Reply:
Maybe we keep it, but can we structure our code so that we try v2 first and packages last and it's likely never executed for most?

Reply:
The structure we support:

```json
{
  "packages": {
    "foo/bar": {
      "1.0.0": {},
      "2.0.0": {},
      "3.0.0": {}
    }
  }
}
```

I think yes, I'll keep it just for compatibility.

But all servers are returning this structure:

```json
{
  "packages": [...]
}
```

I don't know what's inside of this array, it's always empty. Let's just ignore it for now, maybe later someone will come and guide us.